src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1773,6 +1776,37 @@ public void EnsureLastEtagIsPersisted(DocumentsOperationContext context, long do
             return (newEtag, changeVector);
         }
 
+        public static bool FlagsProperlySet(DocumentFlags flags, string changeVector, out string reason)
```

Comment:
I'm worried that in DEBUG we will be doing that. What is the slowdown?

Reply:
Yes, we want that with `Conditional`, I think

Reply:
Sure, but we are using this in Debug only, I'm worried that this can get us severe slowdown during Debug tests, but maybe I'm not right

Reply:
I don't think there is a severe slowdown here... 
I guess the most consuming part is parsing the change-vector and most of our tests has a change-vector of 1 or 2 entries.